git-checkout: add example for checking out files across branches

sethwoodworth opened the pull request.

waldyrious commented:
Thanks! Please use lowercase and snake_case for the tokens, per the guidelines in CONTRIBUTING.md

agnivade commented on pages/common/git-checkout.md:
```diff
@@ -4,12 +4,16 @@
 
 - Switch to another branch:
```
suggestion - "Switch to an existing branch". To match with the next description.

agnivade commented on pages/common/git-checkout.md:
```diff
@@ -4,12 +4,16 @@
 
 `git checkout .`
+
+- Checkout a file from another branch:
```
I would refrain from using the word "checkout" as we are trying to explain what "checkout" means in the first place. How about - "Replace the contents of {{file_name}} in the current branch with {{branch_name}}" ? Or something on those lines ..

waldyrious replied:
@agnivade I agree with your concern but that wording can be a little confusing IMO. How about: "Replace a file in the current working directory with the version of it committed in a given branch"?

agnivade commented:
@sethwoodworth - Any updates on this ?

sethwoodworth commented:
@agnivade Apologies! I had a harddrive failure and lost track of this PR.  I've implemented your and @waldyrious' feedback and pushed a new, rebased commit.

agnivade commented:
@sethwoodworth - oops, hard drive crashes are a pain. Hopefully, nothing too damaging.

Thanks for pushing the updates !
